tsc/meetings/2019-07-11.md

```diff
@@ -0,0 +1,122 @@
+easily extended to support fog volumes and vector fields. Ken mentioned that
+vector fields would be very useful to him. One option is to complete the current
+PR and then introduce a new PR for the extension.
+
+Peter didn't have access to his list of forthcoming work.
+
+Dan:
+
+PR 487 - Fixes CI issue where Houdini builds are skipped rather than fail when
+run outside of ASWF Azure account. Trivial fix, anyone can approve.
+
+PR 483 - Adds code coverage support to CMake, Nick currently reviewing. Minor
+issue where code coverage takes a very long time to run (2 hours+) so can only
+be run sporadically. A number of options to resolve this. Ken suggested running
+the unit tests concurrently, though there may be issues if the tests themselves
+run in parallel. Dan suggested using a smaller data set for some of the heavy
+tests, perhaps with a flag that enables the more expensive tests for a less
+frequent test run. Peter suggested using a mode that he added to the unit tests
+where you can supply a file that defines which unit tests to execute would allow
+for filtering out some of the expensive ones.
+
+All agreed there's lots of maintenance work to do around unit tests, improving
+code coverage, breaking up some of the bigger tests like TestTools, etc.
+
+PR 482 - An attempt to resolve the vulnerabilities reported by SonarCloud. Low
+priority PR, merely included to start conversation. This adds a log or error to
+empty catch statements based on whether they are expected to fail or not. Peter
+commented that this might be problematic for the matrix inverse in particular,
+Dan suggested we slightly refactor this one to not communicate with exceptions.
+
+PR 479 - Meeting notes from last meeting, would quite like a meeting attendee to
+approve before merge.
+
+PR 464 - Unification PR. Peter to look into the spare data API and how to
+retrieve a value previously inserted. Peter prefers exposing a non-const
+OP_Table than communicating the labels to hide down into the OP_Factory, Dan to
+change this.
+
+PR 452 - Dan to address remaining feedback by adding syncNodeVersion support to
+VDB Scatter SOP to be able to add position compression option now that spare
+data API has been introduced.
+
+PR 436 - Delay-loading PR. One remaining question from Peter about what happens
+to the metadata when loading just the metadata in the VDB file as a preliminary
+process to loading the rest of the file. Dan to investigate and reply to this
+then Peter to do a final pass before we merge.
+
+PR 402 - Peter to look into improving VDB SOP documentation, low priority item.
+
+Main upcoming feature includes an extension to point moving to provide merging
+grid functionality. Two options for exposing in Houdini, extending the VDB
+Combine SOP and adding a new VDB Point Merge SOP. Dan presented Nick's argument
+for the latter and all agreed that this is the better option considering compile
+times and the existing node parameterization.
```

Comment:
Disclaimer: I wasn't at this meeting

I'm a bit confused by the comments here which are pro the Combine SOP due to compile times and parameterization. With compile times, I'm fairly confident that splitting it into a new SOP will be more efficient than what you're proposing. The build can be parallelized (given the existing one translation unit=one sop structure) and thus essentially overlapped with the build of the Combine SOP which is **by far** the slowest building node. Shoving more functionality into it is guaranteed to increase the maximum possible build time. Note that as this new functionality will only work on OpenVDB Points grids, only a single grid type will need to be considered, which, given the time complexity of building other nodes, becomes negligible in comparison with more than 1 thread.

In regards to node parameterization, the argument for a new node was to more closely match Houdini workflow for "points" rather than trying to augment a SOP where almost none of it's existing behaviour (maybe none at all?) is actually relevant to OpenVDB Points grids. This will definitely not be immediately intuitive to users wanting to combine (the Houdini terminology for this operation is "merge") OpenVDB Point grids. Note that if we, say, supported activity operations in Houdini on OpenVDB Points grids, this would make sense to live on the Combine node. The act of "merging" however perhaps doesn't. An additional minor point is the nice multi input advantage the merge has vs using an extra Houdini merge.

I almost missed this point, please tag me in future!

Reply:
Yes, Jeff is right, decision was for a separate SOP. With extra clarification:

Two options for exposing in Houdini, extending the VDB Combine SOP and adding a new VDB Point Merge SOP. Dan presented Nick's argument for the latter **(ie a new Point Merge SOP)** and all agreed that this is the better option...

Reply:
Thanks @jmlait @danrbailey, sorry for the confusion!! I'm not sure how I managed to miss-read this :|